samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1884,6 +1897,23 @@ mfxStatus CDecodingPipeline::RunDecoding()
                 bErrIncompatibleVideoParams = true;
                 // need to go to the buffering loop prior to reset procedure
                 pBitstream = NULL;
+                sts = MFX_ERR_NONE;
+                continue;
+            } else if (MFX_ERR_REALLOC_SURFACE == sts) {
+                mfxVideoParam param;
+                m_pmfxDEC->GetVideoParam(&param);
```

Comment:
You never update m_mfxVideoParams. Why?

Reply:
I will change it. Thanks.

Reply:
In the updated latest version, m_mfxVideoParams will be updated in mfxStatus ReallocCurrentSurface(mfxFrameInfo& info). Thanks.